### PR TITLE
Support [LegacyWindowAlias]

### LINKED ISSUE #24474: Support [LegacyWindowAlias]
Needed for [`webkitURL`](https://url.spec.whatwg.org/#webkiturl).

### PR DESCRIPTION
**Pull request by saschanaz**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #24474 (GitHub issue number if applicable)

- [x] There are tests for these changes

### CODE REVIEW

File: components/script/dom/bindings/interface.rs [hunk 1]
```diff
@@ -260,6 +261,12 @@ pub fn create_noncallback_interface_object(
     define_name(cx, rval.handle(), name);
     define_length(cx, rval.handle(), i32::try_from(length).expect("overflow"));
     define_on_global_object(cx, global, name, rval.handle());
+
+    if is_exposed_in(global, Globals::WINDOW) {
```
Comment (nox) [inline components/script/dom/bindings/interface.rs]: This attribute is forbidden in non-Window globals so I'm not sure what this test is for.

**Review by nox — COMMENTED** [review]
Just one question, r=me otherwise. Thanks for your contribution!

File: components/script/dom/bindings/interface.rs (hunk 1 above)
Comment (saschanaz) [reply components/script/dom/bindings/interface.rs]: Interfaces with `[Exposed=(Window, Worker)]` can also have legacy window aliases, while they only actually get aliases on Window. This is to match the specified behavior.

**Comment by saschanaz** [comment]
Sorry to push again, fixed a test-tidy error, interface map data generation (this caused a bug in WebKitCSSMatrix test), and also added WebKitCSSMatrix.

**Comment by nox** [comment]
No worries.

@bors-servo r+

**Comment by jdm** [comment]
This needs `mach update-manifest` now.

File: tests/wpt/mozilla/tests/mozilla/interfaces.js [hunk 2]
```diff
@@ -45,9 +45,9 @@ function test_interfaces(interfaceNamesInGlobalScope) {
       "Uint32Array",
       "Uint8Array",
       "Uint8ClampedArray",
-      "Uint8ClampedArray",
       "WeakMap",
       "WeakSet",
+      "WebKitCSSMatrix",
```
Comment (jdm) [inline tests/wpt/mozilla/tests/mozilla/interfaces.js]: Whoops; this belongs in interfaces.html instead, and with a lowercase K.

Comment (saschanaz) [reply tests/wpt/mozilla/tests/mozilla/interfaces.js]: Whoooops!! Fixed it. It's `WebKitCSSMatrix` anyway though, as WebKit people somehow decided to use uppercase K here.

**Pull request merged**